frontend/src/context/TournamentsContext.tsx

```diff
@@ -108,7 +122,13 @@ export const TournamentsProvider = (): ReactElement => {
       void getAllTournaments();
       isInitialRender.current = false;
     }
-  }, [shouldRefresh]);
+
+    // Fetch the tournaments after creating tournaments or navigatating to location with refresh tag true
+    if (location.state?.refresh) {
+      void getAllTournaments();
+      location.state.refresh = false;
```

Comment:
Just a note; If I am not wrong, directly setting state elements might not be the best practice